WebClinicGUI/Controllers/AccountController.cs

```diff
@@ -256,5 +266,49 @@ public IActionResult ChangeEmail()
         {
             return View();
         }
+
+        [HttpPost]
+        [Authorize]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> ChangeEmail(ChangeEmailViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    var credentials = new ChangeEmailModel(
+                        model.Email,
+                        model.NewEmail,
+                        HashUtils.Hash(model.Password));
+
+                    await _client.SendRequestWithBodyAsync(HttpMethod.Put, "Account/UpdateEmail", credentials);
+
+                    await HttpContext.SignOutAsync();
+
+                    var userDto = new AppUser
+                    {
+                        Email = model.NewEmail,
+                        Password = HashUtils.Hash(model.Password),
+                    };
+                    await Login(userDto);
+
+                    if (_contextAccessor.HttpContext.User.IsInRole("Physician"))
+                        return RedirectToAction("PhysicianAccount");
+                    if (_contextAccessor.HttpContext.User.IsInRole("Receptionist"))
+                        return RedirectToAction("ReceptionistAccount");
+
+                    return RedirectToAction("PatientAccount");
+                }
+                catch (HttpRequestException ex)
+                {
+                    if (ex.Message.Contains("Email \"" + model.NewEmail + "\" is already taken"))
+                        ModelState.AddModelError("WrongCredential", _localizer["EmailExists"]);
+                    else if (ex.Message.Contains("Password is incorrect"))
+                        ModelState.AddModelError("WrongCredential", _localizer["WrongPassword"]);
+                    else ModelState.AddModelError("WrongCredential", _localizer["WrongCredentials"]);

```

Comment:
I'm very sad because of this solution

Reply:
We’ve discussed it in our travel planner app that we shouldnt inform the user what exactly was incorrect - password or e-mail

Reply:
This is a bit different -> you're already logged in. I don't even have to ask for old email